packages/app/src/explorer/EntityListContainer.tsx

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+import { useCallback, useRef } from 'react';
+
+import EntityList from './EntityList';
+import { getActiveButtonIndex } from './utils';
+
+interface Props {
+  selectedPath: string;
+  onSelect: (path: string) => void;
+}
+
+function EntityListContainer(props: Props) {
+  const { selectedPath, onSelect } = props;
+
+  const ref = useRef<HTMLDivElement>(null);
+
+  const focusOnNext = useCallback(() => {
+    const res = getActiveButtonIndex(ref.current);
+    if (!res) {
+      return;
+    }
+
+    const [i, buttonElementList] = res;
+    if (i < 0 || i === buttonElementList.length - 1) {
+      return;
+    }
+    buttonElementList[i + 1].focus();
+  }, []);
+
+  const focusOnPrevious = useCallback(() => {
+    const res = getActiveButtonIndex(ref.current);
+    if (!res) {
+      return;
+    }
+
+    const [i, buttonElementList] = res;
+    if (i <= 0) {
+      return;
+    }
+    buttonElementList[i - 1].focus();
+  }, []);
+
+  return (
+    <div
+      ref={ref}
+      onKeyDown={(evt) => {
+        const { key } = evt;
+
+        if (key === 'ArrowDown') {
+          focusOnNext();
```

Comment:
I'd rename those `focusNext` and `focusPrevious` and return from each `if` block. Not needed but cleaner I think,in case the complexity grows.